DistributedLeaseManager.AzureCosmosDb/ServiceCollectionExtensions.cs

```diff
@@ -11,22 +11,30 @@ public static IServiceCollection AddCosmosDbDistributedLeaseManager(
         this IServiceCollection services,
         string cosmosDbConnectionString,
         string databaseName,
-        string containerName)
+        string containerName,
+        string partitionKeyPath)
     {
         services.TryAddSingleton(new CosmosClient(cosmosDbConnectionString));
 
-        return services.AddCosmosDbDistributedLeaseManager(databaseName, containerName);
+        return services.AddCosmosDbDistributedLeaseManager(databaseName, containerName, partitionKeyPath);
     }
 
     public static IServiceCollection AddCosmosDbDistributedLeaseManager(
         this IServiceCollection services,
         string databaseName,
-        string containerName)
+        string containerName,
+        string partitionKeyPath)
     {
+        if (!partitionKeyPath.StartsWith("/"))
+        {
+            throw new ArgumentException("Must start with /.", nameof(partitionKeyPath));
+        }
```

Comment:
Although unlikely, it is technically possible to have a nested property path for a partition key, i.e. `/company/id`. Perhaps we should add a validation saying it's not currently supported, too

Reply:
Yes, that's a good point.